### PR TITLE
new constant tags for statsd client

### PR DESCRIPTION
**Pull request by jofusa**
I have added a optional param (constant_tags) to the Dogstatsd client to append tags to every metrics call.

Useful for setting project wide tags like version.

### CODE REVIEW

**Comment by jofusa** [comment]
hmm that py26 failure is not happening locally. looks like a set comprehension in mock. anyone else seen this?

**Comment by yannmh** [comment]
Thanks a lot @jofusa, it looks great. We'll review your PR really soon.

Python 2.6 failure seems related with https://github.com/testing-cabal/mock/issues/259. I'll make the necessary to fix it.

**Comment by yannmh** [comment]
https://github.com/DataDog/datadogpy/pull/69 should fix Python 2.6 failure. Would you mind rebasing your PR on top of master please ?

File: tests/unit/dogstatsd/test_statsd.py [hunk 1]
```diff
@@ -113,7 +113,7 @@ def test_tags_and_samples(self):
         def test_tags_and_samples(self):
             for i in range(100):
                 self.statsd.gauge('gst', 23, tags=["sampled"], sample_rate=0.9)
-            t.assert_equal('gst:23|g|@0.9|#sampled')
+            t.assert_equal('gst:23|g|@0.9|#sampled,bar:baz,foo')
```
Comment (yannmh) [inline tests/unit/dogstatsd/test_statsd.py]: I don't think this test will pass, as not part of `TestDogStatsdConstantTags` class.

File: tests/unit/dogstatsd/test_statsd.py [hunk 2]
```diff
@@ -252,6 +252,95 @@ def test_accessing_socket_multiple_times_returns_same_socket(self):
         t.assert_not_equal(FakeSocket(), dogpound.get_socket())
 
 
+class TestDogStatsdConstantTags(object):
```
Comment (yannmh) [inline tests/unit/dogstatsd/test_statsd.py]: I don't think we need to duplicate all tests to check the constant tag feature, as they all use the same codepath (i.e. `_report` method). I would just add, to the existing class test, a single test like:
```python
def test_constant_tags(self):
    self.statsd.constant_tags=...
    self.statsd.gauge('gauge', 123.4)
    assert self.recv() == 'gauge:123.4|g|#bar:baz,foo'
```

What do you think ?

**Comment by jofusa** [comment]
sounds great. will update tests

**Comment by jofusa** [comment]
rebased off master

**Comment by yannmh** [comment]
Thanks @jofusa ! It looks good to me.

Let's merge it so it'll be part of the next release, v0.8.0, planned for next week.

**Pull request merged**